src/Application.php

```diff
@@ -70,6 +90,33 @@ public function __construct(ContainerBuilder $containerBuilder = null, array $va
                 $this['phpdi.callable_resolver']
             );
         });
+
+        // Override the dispatcher with ours to use our event listeners
+        $this['dispatcher'] = $this->share(function () {
+            /**
+             * @var EventDispatcherInterface
+             */
+            $dispatcher = new $this['dispatcher_class']();
+
+            $urlMatcher = new LazyUrlMatcher(function () {
+                return $this['url_matcher'];
+            });
+            if (Kernel::VERSION_ID >= 20800) {
+                $dispatcher->addSubscriber(new RouterListener($urlMatcher, $this['request_stack'], $this['request_context'], $this['logger']));
+            } else {
+                $dispatcher->addSubscriber(new RouterListener($urlMatcher, $this['request_context'], $this['logger'], $this['request_stack']));
+            }
+            $dispatcher->addSubscriber(new LocaleListener($this, $urlMatcher, $this['request_stack']));
+            if (isset($this['exception_handler'])) {
+                $dispatcher->addSubscriber($this['exception_handler']);
+            }
+            $dispatcher->addSubscriber(new ResponseListener($this['charset']));
+            $dispatcher->addSubscriber(new MiddlewareListener($this, $this->callbackInvoker));
+            $dispatcher->addSubscriber(new ConverterListener($this['routes'], $this['callback_resolver'], $this->callbackInvoker));
+            $dispatcher->addSubscriber(new StringToResponseListener());
+
+            return $dispatcher;
+        });
```

Comment:
I feel uneasy copying all that configuration code from Silex :/ If they add something, we will fall behind for sure (I know I wont do a diff regularly, that's too much work).

Here is an idea: extend the container entry (`$container->extend(...)`) to remove the previous MiddlewareListener and replace it by our own? I don't know if we can easily browse the registered event subscribers though… Damn Silex is so not extensible :(